File: numeric/euclidgcd/euclidgcd.rb

def euclidgcd(m, n)
    r = m % n
    while r != 0
        m = n
        n = r
        r = m % n
    end
    return n
end

v_1 = 15
v_2 = 10

if ARGV.length >= 2
    v_1 = ARGV[0].to_i
    v_2 = ARGV[1].to_i
end

puts "#{v_1} #{v_2}"
puts "#{euclidgcd(v_1, v_2)}"

File: numeric/euclidgcd/euclidgcd.exs

defmodule Euclid do
  def gcd(m, 0) do
    m
  end
  def gcd(m, n) do
    gcd(n, rem(m, n))
  end

  def print_all() do
    receive do
      {m, n} -> IO.puts("#{m} #{n}\n#{Euclid.gcd(m, n)}")
    end
  end
end

pid = spawn(fn -> Euclid.print_all() end)

case System.argv() do
  [arg1, arg2] ->
    send(pid, {String.to_integer(arg1), String.to_integer(arg2)})
  _ ->
    send(pid, {15, 10})
end
